typespec-tests/src/main/java/com/cadl/specialheaders/SpecialHeadersClient.java

```diff
@@ -266,4 +443,196 @@ public Resource post(String name) {
+                .toObject(Resource.class);
+    }
+
+    /**
+     * etag headers among other optional query/header/body parameters.
+     *
+     * @param format A sequence of textual characters.
+     * @param filter A sequence of textual characters.
+     * @param timestamp An instant in coordinated universal time (UTC)".
+     * @param body The body parameter.
+     * @param requestConditions Specifies HTTP options for conditional requests based on modification time.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Resource putWithOptionalBody(
+            String format,
+            String filter,
+            OffsetDateTime timestamp,
+            Resource body,
+            RequestConditions requestConditions) {
+        // Generated convenience method for putWithOptionalBodyWithResponse
+        RequestOptions requestOptions = new RequestOptions();
+        String ifMatch = requestConditions == null ? null : requestConditions.getIfMatch();
+        String ifNoneMatch = requestConditions == null ? null : requestConditions.getIfNoneMatch();
+        OffsetDateTime ifUnmodifiedSince = requestConditions == null ? null : requestConditions.getIfUnmodifiedSince();
+        OffsetDateTime ifModifiedSince = requestConditions == null ? null : requestConditions.getIfModifiedSince();
+        if (filter != null) {
+            requestOptions.addQueryParam("filter", filter, false);
+        }
+        if (timestamp != null) {
+            requestOptions.setHeader("timestamp", String.valueOf(timestamp.toEpochSecond()));
+        }
+        if (body != null) {
+            requestOptions.setBody(BinaryData.fromObject(body));
+        }
+        if (ifMatch != null) {
+            requestOptions.setHeader("If-Match", ifMatch);
+        }
+        if (ifNoneMatch != null) {
+            requestOptions.setHeader("If-None-Match", ifNoneMatch);
+        }
+        if (ifUnmodifiedSince != null) {
+            requestOptions.setHeader("If-Unmodified-Since", String.valueOf(new DateTimeRfc1123(ifUnmodifiedSince)));
+        }
+        if (ifModifiedSince != null) {
+            requestOptions.setHeader("If-Modified-Since", String.valueOf(new DateTimeRfc1123(ifModifiedSince)));
+        }
+        return putWithOptionalBodyWithResponse(format, requestOptions).getValue().toObject(Resource.class);
+    }
```

Comment:
Case of etag header with other optional query/header/body parameters. `RequestConditions` is put after the optional body parameter.